### ISSUE TITLE
Use Higher Order Components (HOC) instead of Mixin

### ISSUE THREAD

**Issue by sibelius** [kiddkai/react-native-gestures#26]
HOC can do everything that mixins can, but provides a better API

https://medium.com/@dan_abramov/mixins-are-dead-long-live-higher-order-components-94a0d2f9e750

**Comment by kiddkai** [comment]
Yup, I read this before. This is what react redux is doing right? I guess I will move the API to that if I got more time on this.

**Comment by UADACID** [comment]
@kiddkai sory, can you tell me how to make this lib works?  or maybi projext example who using this lib, sory for my bad english

**Comment by assertchris** [comment]
I am working on a refactor which should fix this issue: https://github.com/kiddkai/react-native-gestures/pull/30
